decimal.go

```diff
@@ -614,20 +614,16 @@ func (d Decimal) Mod(d2 Decimal) Decimal {
 	return d.Sub(d2.Mul(quo))
 }
 
-// Pow returns d to the power d2
 func (d Decimal) Pow(d2 Decimal) Decimal {
 	var temp Decimal
-	if d2.IntPart() == 0 {
-		return NewFromFloat(1)
-	}
-	temp = d.Pow(d2.Div(NewFromFloat(2)))
-	if d2.IntPart()%2 == 0 {
-		return temp.Mul(temp)
-	}
-	if d2.IntPart() > 0 {
-		return temp.Mul(temp).Mul(d)
-	}
-	return temp.Mul(temp).Div(d)
+	x := d.value
+	x1 := d.exp
+	y := d2.value
+	k := x.Exp(x, y, nil)
+	a := big.NewInt(int64(x1)).Mul(big.NewInt(int64(x1)), y)
```

Comment:
As it works for examples you gave it would fail when d2 is a decimal with `exp != 0`. For example 1,15 ^ 3,15 produces
```go
d1 := RequireFromString("1.15")
d2 := RequireFromString("3.15")

d1.Pow(d2)
-> 13177096043800968524.66956658119265657462564607005262011901628595176276300244518482010...
```